### PR TITLE
Add Env Parameter Support

### PR DESCRIPTION
**Pull request by safaci2000**

### CODE REVIEW

File: config-example.js [hunk 1]
```diff
@@ -1,20 +1,20 @@
 module.exports = {
-    "token" : "BOT_TOKEN",
-    "prefix" : "PREFIX_CHARACTER",
+    "token" : process.env.BOT_TOKEN,
+    "prefix" : process.env.PREFIX_CHARACTER || "@",
```
Comment (TobyHorn) [inline config-example.js:3]: Please default the prefix character to `!` and not `@`

File: env.template [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
 MYSQL_ROOT_PASSWORD=example
+MYSQL_USER=user
+MYSQL_PASSWORD=secret
+## You'll need to modify the bootstrap code if you change the DB name
 MYSQL_DATABASE=beaconbot
+MYSQL_HOST=db
+DATABASE_PORT==3306
+
+BOT_TOKEN=token
+PREFIX_CHARACTER=@
```
Comment (TobyHorn) [inline env.template:11]: Prefix should be `!`

File: env.template [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+ADMINISTRATOR_ROLE_NAME=Admins
+SUPER_MODERATOR_ROLE_NAME=Super
```
Comment (TobyHorn) [inline env.template:13]: The "Super" role for GB is "Master Control"

Also be sure that you include the emojis within the names as it looks for exact matches with this version.

File: env.template [hunk 4]
```diff
@@ -1,3 +1,19 @@
...
+MODERATOR_ROLE_NAME=Moderators
+ADMINISTRATOR_CHANNEL_NAME=admins
+SUPER_MODERATOR_CHANNEL_NAME=super
+MODERATOR_CHANNEL_NAME=moderators
+SUPER_MODERATOR_LOG_CHANNEL_NAME=super_log
+MODERATOR_LOG_CHANNEL_NAME=moderator_log
```
Comment (TobyHorn) [inline env.template:19]: moderation-log*

File: env.template [hunk 5]
```diff
@@ -1,3 +1,19 @@
...
+
+BOT_TOKEN=token
+PREFIX_CHARACTER=@
+ADMINISTRATOR_ROLE_NAME=Admins
+SUPER_MODERATOR_ROLE_NAME=Super
+MODERATOR_ROLE_NAME=Moderators
+ADMINISTRATOR_CHANNEL_NAME=admins
+SUPER_MODERATOR_CHANNEL_NAME=super
+MODERATOR_CHANNEL_NAME=moderators
+SUPER_MODERATOR_LOG_CHANNEL_NAME=super_log
```
Comment (TobyHorn) [inline env.template:18]: mc-only-logs*

Comment (TobyHorn) [inline env.template:18]: Master Control*

File: env.template [hunk 6]
```diff
@@ -1,3 +1,19 @@
...
+MYSQL_HOST=db
+DATABASE_PORT==3306
+
+BOT_TOKEN=token
+PREFIX_CHARACTER=@
+ADMINISTRATOR_ROLE_NAME=Admins
+SUPER_MODERATOR_ROLE_NAME=Super
+MODERATOR_ROLE_NAME=Moderators
+ADMINISTRATOR_CHANNEL_NAME=admins
+SUPER_MODERATOR_CHANNEL_NAME=super
```
Comment (TobyHorn) [inline env.template:16]: master-control

File: env.template [hunk 7]
```diff
@@ -1,3 +1,19 @@
...
 MYSQL_DATABASE=beaconbot
+MYSQL_HOST=db
+DATABASE_PORT==3306
+
+BOT_TOKEN=token
+PREFIX_CHARACTER=@
+ADMINISTRATOR_ROLE_NAME=Admins
+SUPER_MODERATOR_ROLE_NAME=Super
+MODERATOR_ROLE_NAME=Moderators
+ADMINISTRATOR_CHANNEL_NAME=admins
```
Comment (TobyHorn) [inline env.template:15]: master-control

File: env.template [hunk 8]
```diff
@@ -1,3 +1,19 @@
...
+SUPER_MODERATOR_CHANNEL_NAME=super
+MODERATOR_CHANNEL_NAME=moderators
```
Comment (TobyHorn) [inline env.template:17]: moderation

File: env.template [hunk 9]
```diff
@@ -1,3 +1,19 @@
...
+MYSQL_USER=user
+MYSQL_PASSWORD=secret
+## You'll need to modify the bootstrap code if you change the DB name
 MYSQL_DATABASE=beaconbot
+MYSQL_HOST=db
+DATABASE_PORT==3306
+
+BOT_TOKEN=token
+PREFIX_CHARACTER=@
+ADMINISTRATOR_ROLE_NAME=Admins
```
Comment (TobyHorn) [inline env.template:12]: Botmaster

File: env.template (hunk 3 above)
Comment (safaci2000) [reply env.template:13]: that's fine.  wasn't planning on merging the full config.

**Pull request merged**